vembrane/ann_types.py

```diff
@@ -140,12 +141,50 @@ def description(self):
 
 class AnnotationListEntry(AnnotationEntry):
     def __init__(
-        self, name: str, sep: str, typefunc: Callable[[str], Any] = None, **kwargs
+        self,
+        name: str,
+        sep: str,
+        typefunc: Callable[[str], Any] = None,
+        inner_typefunc: Callable[[str], Any] = lambda x: x,
+        **kwargs,
     ):
-        typefunc = typefunc if typefunc else lambda v: [x.strip() for x in v.split(sep)]
+        typefunc = (
+            typefunc
+            if typefunc
+            else lambda v: [inner_typefunc(x.strip()) for x in v.split(sep)]
+        )
         super().__init__(name, typefunc, nafunc=lambda: [], **kwargs)
 
 
+class AnnotationListDictEntry(AnnotationEntry):
+    def __init__(self, name: str, **kwargs):
+        def typefunc(x):
+            key_value_tuples = map(lambda v: v.split(":", maxsplit=1), x.split("&"))
+            mapping = defaultdict(list)
+            for key, value in key_value_tuples:
+                mapping[key].append(value)
+            # make sure non-existent keys do not return list() on __getitem__ calls
+            # in the filter expression (without actually building a new dict from
+            # the defaultdict)
+            mapping.default_factory = None
+            return mapping
+
+        super().__init__(name=name, typefunc=typefunc, **kwargs)
+
+
+class AnnotationNumberTotalEntry(AnnotationListEntry):
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name, sep="/", inner_typefunc=int, **kwargs)
+
+
+class AnnotationPredictionScoreEntry(AnnotationEntry):
+    def __init__(self, name: str, **kwargs):
+        def typefunc(x: str) -> Dict[str, float]:
+            return {x[: x.rindex("(")]: float(x[x.index("(") + 1 : x.rindex(")")])}
```

Comment:
```suggestion
            return {x[: x.index("(")]: float(x[x.index("(") + 1 : x.rindex(")")])}
```
(Or use a regex. Regex is probably a little more readable, but I'm okay with both.)

Reply:
... and we could compile it in advance, might do that

Reply:
`(.*)\((.*)\)` should do the trick :D

Reply:
but is it more readable... I don't know.